### PR TITLE
Error test was not using REGION_OVERRIDES

### PR DESCRIPTION
**Pull request by rayluo**
Now we unify a _get_region() to avoid this error in future. A test case was also added.

### CODE REVIEW

File: tests/integration/test_smoke.py [hunk 1]
```diff
@@ -87,14 +86,16 @@
...
     'sns': {'ConfirmSubscription': {'TopicArn': 'a', 'Token': 'b'}},
 }
 
 
-REGION_OVERRIDES = {
-    'devicefarm': 'us-west-2',
-}
+def _get_region(service, default='us-east-1'):
+    return {
+        'devicefarm': 'us-west-2',
```
Comment (jamesls) [inline tests/integration/test_smoke.py]: What's the motivation for removing `REGION_OVERRIDES` and recreating the override dictionary each time _get_region is called?  Why not keep `REGION_OVERRIDES` and just have `_get_region` be: `return REGION_OVERRIDES.get(service, default)`?

File: tests/integration/test_smoke.py [hunk 2]
```diff
@@ -87,23 +86,26 @@
...
     's3': {'ListObjects': {'Bucket': 'thisbucketdoesnotexistasdf'}},
+    'devicefarm': {'GetDevice': {'arn': 'arn:aws:devicefarm:REGION::device:f'}},
     'dynamodb': {'DescribeTable': {'TableName': 'unknowntablefoo'}},
     'sns': {'ConfirmSubscription': {'TopicArn': 'a', 'Token': 'b'}},
 }
 
-
-REGION_OVERRIDES = {
+DEFAULT_REGION = 'us-east-1'
+REGION = {
```
Comment (jamesls) [inline tests/integration/test_smoke.py]: What's the motivation for the rename?  I find `REGION_OVERRIDES` more clear than `REGION`.

File: tests/integration/test_smoke.py [hunk 3]
```diff
@@ -87,23 +86,26 @@
...
     'devicefarm': 'us-west-2',
 }
 
+def _get_client(session, service, region=REGION, default_region=DEFAULT_REGION):
```
Comment (jamesls) [inline tests/integration/test_smoke.py]: Don't have mutable default arguments.

File: tests/integration/test_smoke.py [hunk 4]
```diff
@@ -87,23 +86,26 @@
...
+    region_name=region.get(service, default_region)
```
Comment (jamesls) [inline tests/integration/test_smoke.py]: please follow pep8.

**Comment by kyleknap** [comment]
Looks good. :ship:

**Comment by jamesls** [comment]
Looks good, thanks! Merged in 3408a69b7e1e8ba28c4186124ee186b18979a4a2.

**Pull request closed without merging**